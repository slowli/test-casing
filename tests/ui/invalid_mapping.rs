use test_casing::test_casing;

#[test_casing(2, ["test", "this"].map(String::from))]
fn tested_function(#[map] _arg: &str) {
    // Does nothing
}

#[test_casing(2, ["test", "this"].map(String::from))]
fn other_tested_function(#[map(mut)] _arg: &str) {
    // Does nothing
}

#[test_casing(2, ["test", "this"].map(String::from))]
fn another_tested_function(#[map(ref = "String::as_str")] _arg: &str) {
    // Does nothing
}

fn main() {}
