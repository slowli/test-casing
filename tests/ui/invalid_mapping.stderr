error: expected attribute arguments in parentheses: #[map(...)]
 --> tests/ui/invalid_mapping.rs:4:20
  |
4 | fn tested_function(#[map] _arg: &str) {
  |                    ^^^^^^

error: unknown map transform; only `ref` is supported
 --> tests/ui/invalid_mapping.rs:9:32
  |
9 | fn other_tested_function(#[map(mut)] _arg: &str) {
  |                                ^^^

error: expected identifier
  --> tests/ui/invalid_mapping.rs:14:40
   |
14 | fn another_tested_function(#[map(ref = "String::as_str")] _arg: &str) {
   |                                        ^^^^^^^^^^^^^^^^
